### DISCUSSION TITLE
Explanation of the quest monitor

### DISCUSSION THREAD

**Discussion by ravenfeld** [Helium314/SCEE#578] [General]
@Helium314 
Can you explain this message that you put in the onDestroy

`// stop quest monitor: it should be easy to avoid having it running, because going to settings all the time is annoying`

I was going to make a comment to tell you that it doesn't work to look at the onDestroy toarreter something on the onBack.

But with this comment I'm not sure what you're trying to do.

**Comment by Helium314** [comment]
Sorry, according to git that line was added more than a year ago and I can't remember. Maybe if you dig in history you can find out.

> that it doesn't work to look at the onDestroy toarreter something on the onBack.

I don't understand what you mean.
What is the problem with the quest monitor?

**Comment by ravenfeld** [comment]
Currently you stop the monitor when there is an onDestroy of the activity.
And in the documentation you indicate that the monitor works in the background if you click on Home but if you do onBack it won't work.
This may be true for some phones because they do an onBack then onDestroy but it's not true for all.

So I wanted to know if you were aware of this and if in fact doing onBack should not make the monitor work?
If so, I'll look into it and suggest a fix.

**Reply by Helium314** [reply]
I guess back should make the monitor stop, if it's documented somewhere. But if not then it's ok too, my idea was to have a simple way of disabling the monitor without having to disable the setting. (pressing back, or removing app from recents)

**Comment by ravenfeld** [comment]
Ok, thanks, I'll look into making it work with onBack
